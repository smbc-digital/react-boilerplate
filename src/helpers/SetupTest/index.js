import React from 'react'
import Enzyme, { mount, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

Enzyme.configure({ adapter: new Adapter() })

const useContextMock = jest.fn()
React.useContext = useContextMock

beforeEach(() => {
    useContextMock.mockReset()
})

export { React, mount, shallow, renderer, useContextMock }