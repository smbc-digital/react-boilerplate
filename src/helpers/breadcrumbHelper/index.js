const showBreadcrumbs = (show) =>{
    let breadcrumb = document.getElementById('breadcrumb')
    if(breadcrumb !== null) 
    {
        if(show){
            breadcrumb.classList.remove('hidden')
        }
        else{
            breadcrumb.classList.add('hidden')
        }
    }
}

export default showBreadcrumbs