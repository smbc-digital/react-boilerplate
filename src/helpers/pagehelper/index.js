import json_data from './pagenames.json'

export const getPageRoute = (pageNumber) => {
    let page = json_data.filter(page => page.number === pageNumber)[0]

    return page.route
}