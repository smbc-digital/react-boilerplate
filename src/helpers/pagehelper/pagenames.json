[
	{
		"number": 1,
		"route": "/example/example-page"
	},
	{
		"number": 2,
		"route": "/example/example-page-2"
	}
]