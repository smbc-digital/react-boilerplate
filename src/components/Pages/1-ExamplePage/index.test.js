import { React, mount, useContextMock, renderer } from '../../../helpers/SetupTest'
import ExamplePage from './index'
import { getPageRoute } from '../../../helpers/pagehelper'

describe('ExamplePage', () => {
    const history = { push: jest.fn() }

    it('should render the component', () => {
        // Arrange
        const data = {
            example: {
                value: 'yes',
                isValid: true
            },
            onChange: jest.fn()
        }
        useContextMock.mockReturnValueOnce(data)

        // Act
        let wrapper = mount(<ExamplePage history={history} />)

        // Assert
        expect(wrapper.find('RadioInputsContainer').exists()).toBeTruthy()
        expect(wrapper.find('AlertForm').exists()).toBeTruthy()
        expect(wrapper.find('Button').exists()).toBeTruthy()
        expect(wrapper.find('h1').exists()).toBeTruthy()
        expect(wrapper.find('h2').exists()).toBeTruthy()
        expect(wrapper.find('form').exists()).toBeTruthy()
    })

    it('should call push on submit', () => {
        // Arrange
        const data = {
            example: {
                value: 'yes',
                isValid: true
            },
            onChange: jest.fn()
        }
        useContextMock.mockReturnValueOnce(data)

        // Act
        let wrapper = mount(<ExamplePage history={history} />)
        wrapper.find('form').simulate('submit')


        // Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(1))
    })

    it('should go to page 2 when value is no', () => {
        // Arrange
        const data = {
            example: {
                value: 'no',
                isValid: true
            },
            onChange: jest.fn()
        }
        useContextMock.mockReturnValueOnce(data)

        // Act
        let wrapper = mount(<ExamplePage history={history} />)
        wrapper.find('form').simulate('submit')

        // Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(2))
    })

    describe('snapshot', () => {
		it('should render correctly',() => { 
			// Arrange
            const data = {
                example: {
                    value: 'yes',
                    isValid: true
                },
                onChange: jest.fn()
            }
            useContextMock.mockReturnValueOnce(data)
		
			// Act
			const tree = renderer
			.create(<ExamplePage />)
			.toJSON()
		
			// Assert
			expect(tree).toMatchSnapshot()
		})
	})
})
