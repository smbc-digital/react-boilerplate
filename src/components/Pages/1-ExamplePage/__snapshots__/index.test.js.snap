// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExamplePage snapshot should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Boilerplate
  </h1>
  <h2>
    This is an example form
  </h2>
  <div>
    <section
      className="alert-information-form"
    >
      <span
        className="fa fa-info-circle"
      />
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "This is a boilerplate, enjoy!",
          }
        }
      />
    </section>
  </div>
  <fieldset>
    <legend
      className="sr-only"
    >
      Go to next page?
    </legend>
    <div
      className="input-radio-container"
      role="radiogroup"
    >
      <input
        defaultChecked={true}
        disabled={undefined}
        id="yes"
        name="example"
        onChange={[Function]}
        type="radio"
        value="yes"
      />
      <label
        htmlFor="yes"
      >
        Yes
      </label>
    </div>
    <div
      className="input-radio-container"
      role="radiogroup"
    >
      <input
        defaultChecked={false}
        disabled={undefined}
        id="no"
        name="example"
        onChange={[Function]}
        type="radio"
        value="no"
      />
      <label
        htmlFor="no"
      >
        No
      </label>
    </div>
  </fieldset>
  <button
    className="button-primary"
    disabled={false}
    id={undefined}
    onClick={undefined}
    type="submit"
  >
    Next step
  </button>
</form>
`;
